src/order/order.service.ts

```diff
@@ -33,6 +41,44 @@ export class OrderService {
       updatedAt: new Date(),
     });
 
-    return await createdOrder.save();
+    const session = await this.stripe.checkout.sessions.create({
+      payment_method_types: ['card'],
+      mode: 'payment',
+      line_items: createdOrder.products.map((item) => ({
+        price_data: {
+          currency: 'usd',
+          product_data: {
+            name: item.product.title,
+            description: item.product.description,
+          },
+          unit_amount: item.product.price * 100,
+        },
+        quantity: item.quantity,
+      })),
+      success_url: `${process.env.CLIENT_URL}/orderConfirmation?session_id={CHECKOUT_SESSION_ID}`,
+      cancel_url: `${process.env.CLIENT_URL}/cancel`,
+    });
+
+    createdOrder.checkoutId = session.id;
+
+    await createdOrder.save();
+
+    return {
+      url: session.url,
+      order: createdOrder,
+    };
+  }
+
+  async confirmOrder(userId: string, sessionId: string) {
+    let order = await this.orderModel.findOne({
+      userId,
+      checkoutId: sessionId,
+    });
+
+    if (order) {
+      await this.cartModel.updateOne({ userId }, { $unset: { items: [] } });
+      order.status = OrderStatus.COMPLETED;
+      return await order.save();
```

Comment:
Empty line before return

Reply:
Don`t we need the response if there is no order?

Reply:
@VYaniv , yep, I agree, we need to return an error if the order is not found